apps/passport-server/src/services/generic-issuance/pipelines/LemonadePipeline.ts

```diff
@@ -913,25 +950,32 @@ export class LemonadePipeline implements BasePipeline {
     // ineffective.
     delete ticketCopy.timestampSigned;
     const hash = await getHash(
-      JSON.stringify(ticketCopy) + eddsaPrivateKey + pipelineId
+      JSON.stringify(ticketCopy) + eddsaPrivateKey + pipelineId + ticketPCDType
     );
     return hash;
   }
 
-  private async cacheTicket(ticket: EdDSATicketPCD): Promise<void> {
+  private async cacheTicket(
+    ticket: EdDSATicketPCD | PODTicketPCD
+  ): Promise<void> {
     const key = await LemonadePipeline.getTicketCacheKey(
```

Comment:
the code that generates the cache key for a ticket is duplicated across several files, which may lead to bugs in the future

Reply:
I'll make an issue for fixing this. I wonder if key generation could live in the cache service itself. Currently the cache service doesn't know anything about what it's caching, but having some key generation functions for commonly-stored things like PCDs seems like a good idea.